### PR TITLE
Adds a chat message that shows the layout when starting the raid

### PR DESCRIPTION
**Pull request by Matsyir**
image
Little QOL for Chambers of Xeric so you can keep track of the rotation and know what's next. I usually write it in the chat but this is a bit better. I tested it on 20+ raids like in the picture, writing the rotations beforehand. Included a toggle which is default to on. I'm not a big fan of the text color either but it's to keep it consistent with the rest of the raid's text, and it will contrast with the regular game text once you actually start doing stuff.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java [hunk 1]
```diff
@@ -270,6 +271,57 @@ public void onChatMessage(ChatMessage event)
...
+					// overlay displays it. The layout, then combat & puzzle rooms.
+					String layout = getRaid().getLayout().toCode().replaceAll("#", "").replaceAll("¤", "");
+					ChatMessageBuilder msgBuilder = new ChatMessageBuilder()
+							.append(ChatColorType.HIGHLIGHT)
+							.append("[")
+							.append(layout)
+							.append("]: ");
+
+					boolean firstElement = true;
+					for (Room layoutRoom : getRaid().getLayout().getRooms())
```
Comment (Jasper-ketelaar) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java]: I would personally use streams for this as it makes the code quite a bit easier to read:
```
getRaid().getLayout().getRooms().stream()
			.map(e -> getRaid().getRoom(e.getPosition()))
			.filter(Objects::nonNull)
			.filter(e -> e.getType() == RaidRoom.Type.COMBAT || e.getType() == RaidRoom.Type.PUZZLE)
			.map(e -> e.getType() == RaidRoom.Type.COMBAT ? e.getBoss().getName() : e.getPuzzle().getName())
			.reduce((s1, s2) -> s1 + ", " + s2)
			.ifPresent(msgBuilder::append);
```

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java [hunk 2]
```diff
@@ -81,6 +81,17 @@ default boolean scoutOverlayAtBank()
 
 	@ConfigItem(
 		position = 4,
+		keyName = "layoutChatMsg",
+		name = "Show layout in chat on raid start",
```
[the remark this answers is not in the document]
Comment (Matsyir) [reply runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsConfig.java]: Somewhat agree but I did make sure it was entirely visible without hovering. Initially I didn't put the word "raid" in there but it seemed a bit more clear with it.

**Comment by Kamielvf** [comment]
Shouldn't this just show when the raid gets scouted instead?

**Comment by Matsyir** [comment]
Hmm I could add the option for that, but I think its pointless since the scouting overlay is already present until the start, also it would really spam your chat when looking for a layout

**Comment by Kamielvf** [comment]
People might prefer the text message over the overlay (which you can already toggle off in the settings) so I think it should be done when the raid gets scouted.

My plan is to make the overlay be able to stay active inside of the raid but I have to rewrite some stuff which got me stuck with an error I was unable to fix so far.

**Comment by Matsyir** [comment]
Understandable, once I get home (around Monday, uncertain) I'll add the option for that and implement Jasper's stream solution since it is much nicer. My initial idea was to keep the overlay there aswell but I figured it would be too intrusive, though I see why someone would prefer it over a chat message. I didn't plan on adding whitelist/blacklist colors, or Puzzle/Combat indicators for the unknown rooms (but it can be inferred from the scfpcc.. layout), so as it stands the chat message would not quite be as fully featured as the overlay, but I will look into implementing those as well.

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018, Kamiel
+ * Copyright (c) 2018, Kamiel, Matsyir, Jasper-ketelaar
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java]: New copyrights should be separate entries on new lines.

**Comment by deathbeam** [comment]
Implemented in #7545

**Pull request closed without merging**